재호/17299.py

```diff
@@ -0,0 +1,28 @@
+# https://www.acmicpc.net/problem/17299
+
+from sys import stdin
+
+
+def solution():
+    N = int(stdin.readline())
+    A = list(map(int, stdin.readline().split()))
+    F = [0] * ((10 ** 6) + 1)
+    answer = [0] * (10 ** 6)
```

Comment:
```suggestion
    answer = [-1] * N
```
answer리스트는 A리스트와 같은 길이이면 되므로 길이를 N으로 설정하는게 메모리 측면에서 좋을것 같아요! 그리고 default로 -1로 설정하면 for문을 다 돌고 더 작업을 안해도 되서 효율적인거 같습니다!

Reply:
아 맞네요 answer 리스트를 불필요하게 길게 할당해버렸군요 ㅋㅋ 감사합니다!